src/splitEvery.js

```diff
@@ -0,0 +1,39 @@
+var _curry2 = require('./internal/_curry2');
+var _reduce = require('./internal/_reduce');
+
+
+/**
+ * Subdivides the input array into parts of size `n`.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig Number -> [a] -> [[a]]
+ * @param {Number} n The size of the batches.
+ * @param {Array} list The list to be divide into batches.
+ * @return {Array} The new list.
+ * @aka chunk
+ * @example
+ *
+ *      R.chunk(3, [5,3,8,1,2,9,7]);
+ *      //=> [ [ 5, 3, 8 ],  [ 1, 2, 9 ], [ 7 ] ]
+ */
+
+module.exports = _curry2(function(n, list) {
+  var idx = 0;
+  if (n < 1) {
+    return list.length === 0 ? [] : [list];
```

Comment:
I don't see how streams are different than lists in this regard. If I chunk a stream into parts of size 0 getting the entire stream wrapped in an array doesn't help me. The current behavior treats 0 like i meant "chunk of entire array". That doesn't make sense. If anything should be treated like that it should be Infinity.

My point is: if we can't return anything sensible, so we should throw an error do as little as possible as to signal "nothing I can do here, I'll bail out with this empty list". Otherwise the code pretends to be meaningful when it is not.

We could also return `[[]]`. That would be closer to the requested behavior because there would actually be a chunk of size 0. But again this doesn't _actually_ make sense so the code would again be pretending to make sense and trying to hide a user error.

If we can't justify other libraries behavior I see not reason to duplicate it. That would just be repeating other peoples mistakes.

But if anyone can explain the current behavior in a meaningful way I will rest my case!